WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/CardPresentPaymentsOnboardingUseCase.swift

```diff
@@ -86,6 +91,16 @@ final class CardPresentPaymentsOnboardingUseCase: CardPresentPaymentsOnboardingU
     func updateState() {
         state = checkOnboardingState()
     }
+
+    func selectPlugin(_ plugin: CardPresentPaymentsPlugin) {
+        precondition(state == .selectPlugin)
```

Comment:
I'd swear I had replied to this on Friday 🤔 
It is deliberate, although I've never been super confident when adding `precondition` checks. I don't like the idea of adding the possibility of the app crashing, but on the other hand, the logic here expects this to only be called from the `.selectPlugin` state. Maybe I'll switch it to an `assert`, because the goal is to catch any change in that assumption during development, but I'm not sure keeping that in production is as useful